<?xml version="1.0"?>
<robot name="multipleshapes">

	 <link name="rect_base">
	    <visual>
	      <geometry>
		<box size="0.4 .4 .04"/>
	      </geometry>
		<material name="Greenish">
     			<color rgba="0.25 0.45 0 1"/>
  	       </material>
	    </visual>
	 </link>

	<link name="dell_base">
	    <visual>
	      <geometry>
		<box size="0.22 .37 .02"/>
	      </geometry>
		<material name="Silver">
     			<color rgba="0.95 0.95 1 0.95"/>
  	       </material>
	    </visual>
	 </link>

	<link name="dell_screen">
	    <visual>
	      <geometry>
		<box size="0.22 .37 .01"/>
	      </geometry>
	      <origin rpy="0 1.7854 0" xyz="0 0 0"/>
		<material name="Silver">
     			<color rgba="0.95 0.95 1 0.95"/>
  	       </material>
	    </visual>
	 </link>

	 <link name="bar_left1">
	    <visual>
	      <geometry>
		<box size="0.05 .025 .33"/>
	      </geometry>
		<material name="Gray">
     			<color rgba="0.75 0.75 0.75 1"/>
  	       </material>
	    </visual>
	  </link>

	<link name="bar_right1">
	    <visual>
	      <geometry>
		<box size="0.05 .025 .33"/>
	      </geometry>
		<material name="Gray">
     			<color rgba="0.75 0.75 0.75 1"/>
  	       </material>
	    </visual>
	 </link>

	<link name="wheel_left">
	    <visual>
	      <geometry>
		<cylinder length="0.03" radius="0.05"/>
	      </geometry>
	      <origin rpy="0 1.57075 1.57075" xyz="0 0 0"/>
		<material name="Black">
     			<color rgba="0.1 0.1 0.1 1"/>
  	       </material>
	    </visual>
	 </link>
	<link name="wheel_right">
	    <visual>
	      <geometry>
		<cylinder length="0.03" radius="0.05"/>
	      </geometry>
	      <origin rpy="0 1.57075 1.57075" xyz="0 0 0"/>
		<material name="Black">
     			<color rgba="0.1 0.1 0.1 1"/>
  	       </material>
	    </visual>
	 </link>

	<link name="shaft_left">
	    <visual>
	      <geometry>
		<box size="0.07 .025 .035"/>
	      </geometry>
	      <origin rpy="0 1.7854 0" xyz="0 0 0"/>
		<material name="Silver">
     			<color rgba="0.95 0.95 1 0.95"/>
  	       </material>
	    </visual>
	 </link>


	<link name="shaft_right">
  	    <visual>
	      <geometry>
		<box size="0.07 .025 .035"/>
	      </geometry>
	      <origin rpy="0 1.7854 0" xyz="0 0 0"/>
		<material name="Silver">
     			<color rgba="0.95 0.95 1 0.95"/>
  	       </material>
	    </visual>
	 </link>


	<link name="wheel_left2">
	    <visual>
	      <geometry>
		<cylinder length="0.03" radius="0.025"/>
	      </geometry>
	      <origin rpy="0 1.57075 1.57075" xyz="0 0 0"/>
		<material name="Yellow">
     			<color rgba="0.8 0.8 0 1"/>
  	       </material>
	    </visual>
	 </link>

	<link name="wheel_right2">
	    <visual>
	      <geometry>
		<cylinder length="0.03" radius="0.025"/>
	      </geometry>
	      <origin rpy="0 1.57075 1.57075" xyz="0 0 0"/>
		<material name="Yellow">
     			<color rgba="0.8 0.8 0 1"/>
  	       </material>
	    </visual>
	 </link>

	<link name="bar_ver1">
	    <visual>
	      <geometry>
		<cylinder length="0.3" radius="0.01"/>
	      </geometry>
	      <origin rpy="0 1.57075 0" xyz="0 0 0"/>
		<material name="Gray">
     			<color rgba="0.75 0.75 0.75 1"/>
  	       </material>
	    </visual>
	 </link>

	<link name="bar_ver2">
	    <visual>
	      <geometry>
		<cylinder length="0.3" radius="0.01"/>
	      </geometry>
	      <origin rpy="0 1.57075 0" xyz="0 0 0"/>
		<material name="Gray">
     			<color rgba="0.75 0.75 0.75 1"/>
  	       </material>
	    </visual>
	 </link>

	<link name="bar_hor1">
	    <visual>
	      <geometry>
		<box size="0.05 .4 .02"/>
	      </geometry>
		<material name="Gray">
     			<color rgba="0.75 0.75 0.75 1"/>
  	       </material>
	    </visual>
	 </link>

	<link name="bar_hor2">
	    <visual>
	      <geometry>
		<box size="0.05 .4 .02"/>
	      </geometry>
		<material name="Gray">
     			<color rgba="0.75 0.75 0.75 1"/>
  	       </material>
	    </visual>
	 </link>


	<link name="bar_hor3">
	    <visual>
	      <geometry>
		<box size="0.05 .4 .02"/>
	      </geometry>
		<material name="Gray">
			<color rgba="0.75 0.75 0.75 1"/>
	       </material>
	    </visual>
	 </link>

	<link name="bar_hor4">
	    <visual>
	      <geometry>
		<box size="0.05 .4 .02"/>
	      </geometry>
		<material name="Gray">
			<color rgba="0.75 0.75 0.75 1"/>
	       </material>
	    </visual>
	 </link>

	<link name="bar_right2">
	    <visual>
	      <geometry>
		<cylinder length="0.12" radius="0.015"/>
	      </geometry>
		<material name="Gray">
     			<color rgba="0.75 0.75 0.75 1"/>
  	       </material>
	    </visual>
	 </link>

	<link name="bar_left2">
	    <visual>
	      <geometry>
		<cylinder length="0.12" radius="0.015"/>
	      </geometry>
		<material name="Gray">
     			<color rgba="0.75 0.75 0.75 1"/>
  	       </material>
	    </visual>
	 </link>

	<link name="kinect_link">
	  <visual>
	    <geometry>
	      <box size="0.064 0.2 0.0381" />
	    </geometry>
	    <material name="Black" />
	  </visual>
	  <inertial>
	      <mass value="0.0001" />
	      <origin xyz="0 0 0" />
	      <inertia ixx="0.0001" ixy="0.0" ixz="0.0"
		       iyy="0.0001" iyz="0.0" 
		       izz="0.0001" />
	  </inertial>
	</link>

	 <joint name="base_to_dell" type="fixed">
	    <parent link="rect_base"/>
	    <child link="bar_left1"/>
	    <origin xyz="0.175 0.1875 .0925"/>
	 </joint>

	 <joint name="base_to_bar_left1" type="fixed">
	    <parent link="rect_base"/>
	    <child link="dell_base"/>
	    <origin xyz="-0.05 0 .12"/>
	 </joint>

	 <joint name="base_to_bar_right1" type="fixed">
	    <parent link="rect_base"/>
	    <child link="bar_right1"/>
	    <origin xyz="0.175 -0.1875 .0925"/>
	 </joint>

	<joint name="base_to_bar_left2" type="fixed">
	    <parent link="rect_base"/>
	    <child link="bar_right2"/>
	    <origin xyz="-0.1 0.185 0.05"/>
	 </joint>

	<joint name="base_to_bar_right2" type="fixed">
	    <parent link="rect_base"/>
	    <child link="bar_left2"/>
	    <origin xyz="-0.1 -0.185 0.05"/>
	 </joint>

	 <joint name="base_to_bar_ver1" type="fixed">
	    <parent link="rect_base"/>
	    <child link="bar_ver1"/>
	    <origin xyz="0.025 0.185 .1"/>
	 </joint>

	<joint name="base_to_bar_ver2" type="fixed">
	    <parent link="rect_base"/>
	    <child link="bar_ver2"/>
	    <origin xyz="0.025 -0.185 .1"/>
	 </joint>

	 <joint name="base_to_bar_hor1" type="fixed">
	    <parent link="rect_base"/>
	    <child link="bar_hor1"/>
	    <origin xyz="0.175 0 .25"/>
	 </joint>

	<joint name="base_to_bar_hor2" type="fixed">
	    <parent link="rect_base"/>
	    <child link="bar_hor2"/>
	    <origin xyz="0.175 0 .1"/>
	 </joint>

	<joint name="base_to_bar_hor3" type="fixed">
	    <parent link="rect_base"/>
	    <child link="bar_hor3"/>
	    <origin xyz="0.0 0 .1"/>
	 </joint>

	<joint name="base_to_bar_hor4" type="fixed">
	    <parent link="rect_base"/>
	    <child link="bar_hor4"/>
	    <origin xyz="-0.1 0 .1"/>
	 </joint>


	<joint name="base_to_kinect" type="fixed">
	    <parent link="rect_base"/>
	    <child link="kinect_link"/>
	    <origin xyz="0.175 0 0.285"/>
	 </joint>

	 <joint name="base_to_wheel_left" type="fixed">
	    <parent link="rect_base"/>
	    <child link="wheel_left"/>
	    <origin xyz="0.175 0.215 -0.05"/>
	 </joint>

	<joint name="base_to_wheel_right" type="fixed">
	    <parent link="rect_base"/>
	    <child link="wheel_right"/>
	    <origin xyz="0.175 -0.215 -0.05"/>
	 </joint>

         <joint name="base_to_wheel_left1" type="fixed">
	    <parent link="rect_base"/>
	    <child link="wheel_left2"/>
	    <origin xyz="-0.175 0.185 -0.075"/>
	 </joint>

	<joint name="base_to_wheel_right1" type="fixed">
	    <parent link="rect_base"/>
	    <child link="wheel_right2"/>
	    <origin xyz="-0.175 -0.185 -0.075"/>
	 </joint>

         <joint name="base_to_shaft_left1" type="fixed">
	    <parent link="rect_base"/>
	    <child link="shaft_left"/>
	    <origin xyz="-0.16 0.185 -0.025"/>
	 </joint>

	<joint name="base_to_shaft_right1" type="fixed">
	    <parent link="rect_base"/>
	    <child link="shaft_right"/>
	    <origin xyz="-0.16 -0.185 -0.025"/>
	 </joint>

	<joint name="dell_base_to_screen" type="fixed">
	    <parent link="dell_base"/>
	    <child link="dell_screen"/>
	    <origin xyz="0.11 0 0.11"/>
	 </joint>


</robot>
